pubnub-gson/src/main/java/com/pubnub/internal/v2/subscription/SubscriptionImpl.kt

```diff
@@ -51,4 +69,85 @@ class SubscriptionImpl(
             }
         }
     }
+
+    override fun setOnMessage(onMessageHandler: OnMessageHandler?) {
+        setOnMessageHandler(onMessageHandler)
+    }
+
+    override fun setOnSignal(onSignalHandler: OnSignalHandler?) {
+        setOnSignalHandler(onSignalHandler)
+    }
+
+    override fun setOnPresence(onPresenceHandler: OnPresenceHandler?) {
+        setOnPresenceHandler(onPresenceHandler)
+    }
+
+    override fun setOnMessageAction(onMessageActionHandler: OnMessageActionHandler?) {
+        setOnMessageActionHandler(onMessageActionHandler)
+    }
+
+    override fun setOnUuidMetadata(onUuidMetadataHandler: OnUuidMetadataHandler?) {
+        setOnUuidMetadataHandler(onUuidMetadataHandler)
+    }
+
+    override fun setOnChannelMetadata(onChannelMetadataHandler: OnChannelMetadataHandler?) {
+        setOnChannelMetadataHandler(onChannelMetadataHandler)
+    }
+
+    override fun setOnMembership(onMembershipHandler: OnMembershipHandler?) {
+        setOnMembershipHandler(onMembershipHandler)
+    }
+
+    override fun setOnFile(onFileHandler: OnFileHandler?) {
+        setOnFileHandler(onFileHandler)
+    }
+
+    private fun setOnMessageHandler(onMessageHandler: OnMessageHandler?) {
+        emitterHelper.onMessage = onMessageHandler?.let { handler ->
```

Comment:
if you change the type in emitterHelper to OnMessageHandler, this could be just
`emitterHelper.onMessage = onMessageHandler`

(see comment in EmitterHelper)

Reply:
Got it.